annotations/s2i-annotations/src/main/java/io/dekorate/s2i/decorator/AddBuilderImageStreamResourceDecorator.java

```diff
@@ -43,8 +43,9 @@ public AddBuilderImageStreamResourceDecorator(S2iBuildConfig config) {
   }
 
   public void visit(KubernetesListBuilder list) {
-    ObjectMeta meta = getMandatoryDeploymentMetadata(list);
-
+    List<ObjectMeta> metaList = getDeploymentMetadataList(list, this.config.getName());
+    //If there are multiple deployment resource we can't possibly determine which one to use.
+    ObjectMeta meta = metaList.size() == 1 ? metaList.get(0) : new ObjectMeta();
```

Comment:
These three lines are equivalent to simply do:
`getDeploymentMetadata(list, this.config.getName()).orElseGet(ObjectMeta::new);`